packages/frontend/src/features/comments/components/CommentForm.tsx

```diff
@@ -19,27 +22,38 @@ export const CommentForm: FC<Props> = ({
     onCancel,
     mode = 'new',
 }) => {
-    const commentForm = useForm<Pick<Comment, 'text' | 'replyTo'>>({
+    const { data: listUsers, isSuccess } = useOrganizationUsers();
+    let userNames: SuggestionsItem[] = useMemo(
+        () =>
+            listUsers?.map((user) => ({
+                label: user.firstName + ' ' + user.lastName,
+                id: user.userUuid,
+            })) || [],
+
+        [listUsers],
+    );
+    const [mentions, setMentions] = useState<string[]>([]);
+    console.log({ mentions });
+
+    const [commentText, setCommentText] = useState('');
+    const [commentHtml, setCommentHtml] = useState('');
+    const [shouldClearEditor, setShouldClearEditor] = useState(false);
+
+    const commentForm = useForm<Pick<Comment, 'replyTo'>>({
         initialValues: {
-            text: '',
             replyTo: '',
         },
-        validate: {
-            text: (value) => {
-                if (value.trim() === '') {
-                    return 'Comment cannot be empty';
-                }
-                return null;
-            },
-        },
     });
-    const [mentions, setMentions] = useState<string[]>([]);
 
-    const handleSubmit = commentForm.onSubmit(async ({ text }) => {
-        await onSubmit(text, mentions);
-        setMentions([]);
+    const handleSubmit = commentForm.onSubmit(async () => {
+        if (commentText === '') return;
 
-        commentForm.reset();
+        // get comment text - commentText
+        // get comment html - commentHtml
+        // get mentions - mentions
+        // submit comment with these 3 values: text, html, mentions
+        // await onSubmit(commentText);
+        setShouldClearEditor(true);
```

Comment:
we should submit the form with 3 things: 
1. plain text (in case we need to refer to it in the future/utilise it for comment previews)
2. html (to render the comment with nice formatting, accepting multiline)
3. mentions (an array of strings that will be useful when displaying notification data)

Some people recommend storing the JSON as well - which would be handy for edit functionality. 

https://github.com/ueberdosis/tiptap/discussions/148#discussioncomment-7319

what do you think?

Reply:
tl;dr let's store plain_text and html :muscle: 

yeah, storing html makes sense, because that contains information that can't be generated from the plain text.
I don't think we need json... I'm thinking on emails, html and plain text is all you need. 

If we need to pick one, I'd choose storing html over plain text, because you can generate plain text from html by removing the html tags, but not the other way around.

Reply:
>If we need to pick one, I'd choose storing html over plain text, because you can generate plain text from html by removing the html tags, but not the other way around.

great point, didn't think of that. 
Also, `tiptap` provides utils to parse this content on all types: `text`, `html`, and `json`